File: 58-length-of-last-word/length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        boolean wordFound = false;
        int counter = 0;
        for (int i = s.length() - 1; i >= 0; i--) {
            if (s.charAt(i) != ' ') {
                counter++;
                wordFound = true;
            } else if (wordFound == true) {
                break;
            }
        }
        return counter;
    }
}

File: 58-length-of-last-word/length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        last_word = ""
        new_word = True
        for c in s:
            if c != " " and new_word:
                last_word = c
                new_word = False
            elif c != " ":
                last_word += c
            else:
                new_word = True
        return len(last_word)